stl/inc/algorithm

```diff
@@ -1567,6 +1567,39 @@ namespace ranges {
     };
 
     inline constexpr _Copy_n_fn copy_n{_Not_quite_object::_Construct_tag{}};
+
+    // ALIAS TEMPLATE copy_backward_result
+    template <class _In, class _Out>
+    using copy_backward_result = in_out_result<_In, _Out>;
+
+    // VARIABLE ranges::copy_backward
+    class _Copy_backward_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <bidirectional_iterator _It1, sentinel_for<_It1> _Se1, bidirectional_iterator _It2>
+            requires indirectly_copyable<_It1, _It2>
+        constexpr copy_backward_result<_It1, _It2> operator()(_It1 _First, _Se1 _Last, _It2 _Result) const {
+            _Adl_verify_range(_First, _Last);
+            const auto _UFirst = _Get_unwrapped(_STD move(_First));
+            auto _ULast        = _Get_final_iterator_unwrapped<_It1>(_UFirst, _STD move(_Last));
+            _Seek_wrapped(_First, _ULast);
+            _Result = _Copy_backward_unchecked(_STD move(_UFirst), _STD move(_ULast), _STD move(_Result));

```

Comment:
Cant we use `_Get_unwrapped_unverified(_STD move(_Result))` here?

Reply:
This will eventually use `_Get_unwrapped_n(_Result, _Idl_distance(_UFirst, _ULast))`. While we *could* use `_Get_unwrapped_unverified` without harming correctness I prefer to avoid unwrapping completely to make the things that will need fixing later easier to spot.

Reply:
I just rewrote the issue description in #893 with some details of the iterator unwrapping machinery that hopefully clarify my response.

Reply:
Thanks a lot :+1: